fhir-profile/src/main/java/com/ibm/fhir/profile/ConstraintGenerator.java

```diff
@@ -161,15 +171,17 @@ private String generate(Node node) {
         }
 
         if (hasVocabularyConstraint(elementDefinition)) {
-            return generateVocabularyConstraint(elementDefinition);
+            String expr = generateVocabularyConstraint(elementDefinition);
+            if (node.children.stream().noneMatch(child -> hasConstraint(child))) {
+                return expr;
+            }
+            sb.append(expr).append(" and ");
```

Comment:
will there ALWAYS be another constraint following this and?

Reply:
if hasConstraint returns true on the check above it, then yes.